lib/asciidoctor/opal_ext/file.rb

```diff
@@ -19,12 +19,19 @@ class File
   attr_reader :lineno
   attr_reader :path
 
-  def initialize(path, mode = 'r')
+  def initialize(path, flags = 'r')
     @path = path
     @contents = nil
     @eof = false
     @lineno = 0
-    @fd = `require('fs').openSync(path, mode)` if JAVASCRIPT_PLATFORM == 'node'
+    # binary flag is unsupported
+    binary_flag_regexp = /b/
```

Comment:
> If you read the image file with the encoding set to 'utf8', you get corrupt data. (I'm not sure if that's what the code is doing. I'm just mentioning that a general comment).

I think you are right, the Ruby code is:
```ruby
bindata = ::File.open(image_path, 'rb') {|file| file.read }
%(data:#{mimetype};base64,#{::Base64.encode64(bindata).delete EOL})
```

So basically we are instantiating a new File with `initialize` and then we are using `File.read` which resolves to:

```javascript
return require('fs').readFileSync(path, 'utf8');
```

Maybe we should implement `IO.binread` ? In this case we will use the following code, instead of `File.read`:
```ruby
bindata = ::IO.binread(image_path)
```

[abstract_node.rb#L367](https://github.com/asciidoctor/asciidoctor/blob/a16566c34e51e9f1078b67374e2869281a040205/lib/asciidoctor/abstract_node.rb#L367)